### DISCUSSION TITLE
Build Assets using Gulp

### DISCUSSION THREAD

**Discussion by hishamco** [OrchardCMS/OrchardCore#12012] [General]
@Skrypt I'm working on fixing UI issue, but again I'm suffering from building assets, is there any guidelines?

I already run `npm install` but there's an issue when `install:orchardcore.apis.graphql` is executed. FYI I already upgrade both NodeJS & NPM

**Comment by ns8482e** [comment]
@hishamco any specific error are you getting?

**Reply by hishamco** [reply]
I think we need to document the proper flow to build the assets. May be it's a long time for me to chane assets in any module. Assume I need to change TheAdmin assets, what are the commands I need to run? Do I need to clean anything before?

**Reply by Skrypt** [reply]
I'm currently using Node 16.14.2 and npm 8.5.0.
If you are having issues with these after cleaning the project with `git clean -xdf` then let us know.

**Reply by hishamco** [reply]
I'm not sure why I need to do `git clean -xdf` frequently ;)

**Reply by Skrypt** [reply]
Because Node package management is a mess? 😉

**Reply by hishamco** [reply]
@ns8482e I'm getting an error when installing GraphQL

![Npm](https://user-images.githubusercontent.com/3237266/180026898-6be23ead-615a-4c37-842a-936ec988482c.png)

**Reply by ns8482e** [reply]
I am on node -14.3 and npm 6.14.8 and it works

**Reply by ns8482e** [reply]
I am running `npm install` on root  of orchardcore clone

**Reply by hishamco** [reply]
I removed `package.lock.json` for each module, then I used `npm install --force` everything seems works again

**Comment by hishamco** [comment]
Similar issue occurs when I used Yarn :(

**Reply by jptissot** [reply]
Deleting node_modules usually fixes it for me. You can also try to delete the package.lock.json, that might work.

**Reply by hishamco** [reply]
Still I'm getting the same error in https://github.com/OrchardCMS/OrchardCore/discussions/12012#discussioncomment-3190567
